Aam-Digital/ndb-core
# Enable Matching Component to also handle multi-select / entity-array cases

Extends the MatchingEntitiesComponent to support "group matchings":
- whether group / multiSelect is possible is inferred from the entity schema of the property defined in the `onMatch` config for the matching to be created
- multiSelect mode allows to highlight multiple rows in the selection table and create a match filling the entity-array datatype property
- single select continues to work as previously

To test:
- run locally and switch ChildSchoolRelation.childId datatype from "entity" to "entity-array" (leave the typescript type as `string` - our fallbacks can handle that)
- to see the group matchings in the Education History list correctly, the ChildSchoolOverview also needs to be replace by RelatedTimePeriodEntities component
